### PR TITLE
add types for @mplavin-tools/pub-demo-ts lib

### PR DESCRIPTION
**Pull request by MaxPlav**
Please fill in this template.

- [ ] Use a meaningful title for the pull request. Include the name of the package modified.
- [ ] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test YOUR_PACKAGE_NAME`.)
- [ ] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [ ] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [ ] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If adding a new definition:
- [ ] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [ ] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [ ] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [ ] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [ ] `tslint.json` should be present and it shouldn't have any additional or disabling of rules. Just content as `{ "extends": "dtslint/dt.json" }`. If for reason the some rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]`  and not for whole package so that the need for disabling can be reviewed.
- [ ] `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: <<url here>>
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] Include [tests for your changes](https://github.com/DefinitelyTyped/DefinitelyTyped#testing)
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

If removing a declaration:
- [ ] If a package was never on Definitely Typed, you don't need to do anything. (If you wrote a package and provided types, you don't need to register it with us.)
- [ ] Delete the package's directory.
- [ ] Add it to `notNeededPackages.json`.

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## mplavin-tools__pub-demo-ts ([<kbd>unpkg</kbd>](https://unpkg.com/browse/mplavin-tools__pub-demo-ts@latest/))
was missing the following properties:
1. circle
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 95c35d1cb1f220010395da66df48b303a78a343b
</p>

**Comment by amcasey** [comment]
@MaxPlav Am I correct in guessing that you are also the author of the underlying library?  Normally, when a package is authored in TS, we encourage the package authors to publish the types themselves, so that they're always in sync with the code.  That doesn't work for everyone though, so another common option is to set up (e.g.) a GH action (I assume GitLab has something similar) to push new types to DT whenever the package is updated.

Having said that, if you prefer to have manually maintained types hosted on DT, it would be nice to have tests.  I'm not sure why you deleted yours.  I grabbed your code and they seem to compile with a few minor edits:

```ts
import lib = require('@mplavin-tools/pub-demo-ts');

const circle = new lib.Circle(5);
```

Please let me know how you'd like to proceed.

File: types/mplavin-tools__pub-demo-ts/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,35 @@
+// Type definitions for @mplavin-tools/pub-demo-ts 0.1
+// Project: https://gitlab.com/mplavin/pub-demo-ts
+// Definitions by: Max Plavinskiy <https://github.com/MaxPlav>
+// Definitions: https://github.com/DefinitelyTyped/DefinitelyTyped
+
+    export interface RoundInterface {
```
Comment (amcasey) [inline types/mplavin-tools__pub-demo-ts/index.d.ts:6]: Whitespace (from formerly being in a namespace?)

**Review by amcasey — CHANGES_REQUESTED** [review]

**Comment by MaxPlav** [comment]
> so another common option is to set up (e.g.) a GH action (I assume GitLab has something similar) to push new types to DT whenever the package is updated.
@amcasey, thank you for your notes here. Could you point me to docs or examples of how ppl push new types automatically when a package is released? That's my first try at publishing any TS types.

**Comment by amcasey** [comment]
@MaxPlav I'm not aware of any docs, since it's not a process we officially support - from our perspective, some users just create very large PRs. 😉  I looked for an example for you, but they're hard to track down since the special code is in the originating repo - not this one.

I can tell you that the GH API you'd need to call to automatically create a PR is most conveniently consumed via [Octokit](https://developer.github.com/v3/libraries/).  GitLab appears to have [CI/CD](https://about.gitlab.com/stages-devops-lifecycle/continuous-integration/), which is how you would trigger calls to your Octokit code.

Sorry I don't have a better answer for you.  If I find an example, I'll post an update.

**Pull request closed without merging**